src/controller_plugin.py

```diff
@@ -242,6 +243,65 @@ def get_file(self, file_name, encoding=None):
             return content.decode(encoding)
         return content
 
+    def track_job(self, job):
+        """ Tracks a job by location to prevent request_job_for_client from
+            returning new jobs for locations being tracked. Raises a value
+            error if attempting to insert a job that has a location that is
+            already tracked.
+
+        Arguments:
+            job {dict} -- A job for which its location will be tracked until
+                          respond output is called or the tracking is cleared
+        """
+
+        if self.job_location(job) not in self.tracked_jobs:
+            self.tracked_jobs[self.job_location(job)] = job
+        else:
+            raise ValueError("The location: {} is already being tracked."
+            .format(self.job_location(job)))
+
+    def clear_tracking(self, location, is_error=False):
```

Comment:
Is this called by the plugin author? might be better to clear it automatically when respond_output/error is called.

Reply:
clear tracking is meant to be used to cancel location tracking abruptly due to error or other non standard situation. untrack is to be used for normal removal from the tracking object. should respond output call untrack? or keep the entire system opt in and have authors call it